mcs/class/corlib/Test/System.IO/FileTest.cs

```diff
@@ -487,7 +486,6 @@ public void Delete_Directory_DoesNotExist ()
 
 			try {
 				File.Delete (path);
-				Assert.Fail ("#1");
```

Comment:
I am not sure I quite understand the logic here - removing the `Assert.Fail` only makes sense if the previous `File.Delete` could possibly succeed - and keeping the `catch` only makes sense if it might possibly fail.  Is there a specific reason why this could happen?

Could you please also add some comment to the `try` cause why there shouldn't be an `Assert.Fail`?